tools/shared/src/main/scala/org/scalajs/core/tools/linker/backend/emitter/JSDesugaring.scala

```diff
@@ -195,15 +195,14 @@ import java.io.StringWriter
  *
  *  @author Sébastien Doeraene
  */
-private[emitter] class JSDesugaring(semantics: Semantics,
-    outputMode: OutputMode, internalOptions: InternalOptions) {
+private[emitter] class JSDesugaring(semantics: Semantics, outputMode: OutputMode,
+    jsGen: JSGen) {
```

Comment:
If we give `semantics` and `outputMode` to `JSGen` and make them public `val`s thereof, then `JSDesugaring` and `ClassEmitter` only need to take a `JSGen` as parameter. It also guarantees some more consistency in the semantics and output mode applied between a `JSGen`'s methods and other parts using the `JSGen`